komorebi/bubblemenu/item.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from enum import IntEnum
+
+from gi.repository import Clutter, Gdk
+
+# View modes for a menu item
+class ViewMode(IntEnum):
+	VISIBLE = 0		# Item is visible and interactive
+	GREYED = 1		# Item is visible but "greyed" out, can't be interacted with
+	INVISIBLE = 2	# Item is invisible and doesn't occupy space on the menu
+
+class BubbleMenuItem(Clutter.Text):
+	def __init__(self, text : str, callback):
+		Clutter.Text.__init__(self)
+		logging.debug(f'Initializing BubbleMenuItem(text="{text}")')
+
+		# Setup properties
+		self.set_x_align(Clutter.ActorAlign.START)
+		self.set_x_expand(True)
+		self.set_reactive(False)
+		self.set_selectable(False)
+		self.set_margin_top(5)
+
+		self.set_font_name('Lato 15')
+		self.set_text(text)
+		self.set_color(Clutter.Color.from_string('white')[1])
+
+		self.signals_setup()
+
+		self.hide()
+
+		self.connect_after("button_press_event", callback)
+		logging.debug('Initialized BubbleMenuItem')
+
+	def signals_setup(self):
+		def _on_button_press_event(self, e):
+			self.set_opacity(100)
+
+			return False
```

Comment:
I assume these returns have meaning right?

nitpick: Please don't add a newline before the returns here, it makes the code harder to read (makes it harder to identify blocks) and the functions aren't big enough to warrant that.

Reply:
Yes, although they look "random", they are used for the signal logic on **GTK**, indicating whether this signal should be propagated further or not. Although the other two aren't used anywhere else and can terminate early, this one **can't**, as it's used on upper-levels to implement `MenuItem`'s